File: java/linkedList/MyCycle.java
package linkedList;

public class MyCycle {
    MyLinkedList list;

    public MyCycle (MyLinkedList list) {
        this.list = list;
    }

    public void createCycle(int[] nums, int pos) {
        for (int num : nums) {
            list.addAtTail(num);
        }
        if (nums.length <= 1 || pos < 0 || pos >= nums.length - 1) {
            return;
        }
        ListNode node1 = list.head;
        for (; node1.next != null; node1 = node1.next) {}
        ListNode node2 = list.head;
        for (int i = 0; i < nums.length; i++) {
            if (i == pos) {
                node1.next = node2;
                return;
            }
            node2 = node2.next;
        }
    }

    public void visitCycle() {
        ListNode cur = list.head;
        int size = list.size;
        String str = "";
        for (int i = 0; i < size; i++) {
            str += cur.val + (i == size - 1 ? "" : " -> ");
            cur = cur.next;
        }
        System.out.println(str);
    }
}

File: javascript/linkedList/MyCycle.js
import { MyLinkedList } from './MyLinkedList.js';

var MyCycle = function(list) {
    this.list = list;
}

MyCycle.prototype.createCycle = function(nums, pos) {
    for (let num of nums) {
        this.list.addAtTail(num);
    }
    if (nums.length <= 1 || pos < 0 || pos >= nums.length - 1) {
        return;
    }
    let node1 = this.list.head;
    for (; node1.next; node1 = node1.next) {}
    let node2 = this.list.head;
    for (let i = 0; i < nums.length; i++) {
        if (i == pos) {
            node1.next = node2;
            return;
        }
        node2 = node2.next;
    }
}

MyCycle.prototype.visitCycle = function() {
    let cur = this.list.head;
    let size = this.list.size;
    let str = '';
    for (let i = 0; i < size; i++) {
        str += cur.val + (i != size - 1 ? ' -> ' : '');
        cur = cur.next;
    }
    console.log(str);
}

export { MyCycle };
